internal/webhooks/resources.go

```diff
@@ -0,0 +1,116 @@
+package webhooks
+
+import (
+	"context"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/semconv"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func serviceName(pod corev1.Pod, resources map[string]string) string {
+	if name := resources[string(semconv.K8SDeploymentNameKey)]; name != "" {
+		return name
+	}
+	if name := resources[string(semconv.K8SStatefulSetNameKey)]; name != "" {
+		return name
+	}
+	if name := resources[string(semconv.K8SJobNameKey)]; name != "" {
+		return name
+	}
+	if name := resources[string(semconv.K8SCronJobNameKey)]; name != "" {
+		return name
+	}
+	if name := resources[string(semconv.K8SPodNameKey)]; name != "" {
+		return name
+	}
+	return pod.Spec.Containers[0].Name
+}
+
+// createResourceMap creates resource attribute map.
+// User defined attributes (in explicitly set env var) have higher precedence.
+func (h *handler) createResourceMap(ctx context.Context, extraAttrs map[string]string, ns corev1.Namespace, pod corev1.Pod) map[string]string {
+	// get existing resources env var and parse it into a map
+	existingRes := map[string]bool{}
+	existingResourceEnvIdx := getIndexOfEnv(pod.Spec.Containers[0].Env, envOTELResourceAttrs)
+	if existingResourceEnvIdx > -1 {
+		existingResArr := strings.Split(pod.Spec.Containers[0].Env[existingResourceEnvIdx].Value, ",")
+		for _, kv := range existingResArr {
+			keyValueArr := strings.Split(strings.TrimSpace(kv), "=")
+			if len(keyValueArr) != 2 {
+				continue
```

Comment:
This must be invalid part of the `OTEL_RESOURCE_ATTRIBUTES` value, right? Maybe worth throwing a warning log here?